app/models/user.rb

```diff
@@ -14,13 +13,19 @@ def password_complexity
 
     # Passwords must have characters from at least two groups, identified by these regexes (last one is punctuation)
     matches = [/[a-z]/, /[A-Z]/, /[0-9]/, /[^\w\s]/].select { |rx| rx.match(password) }.size
-    errors.add :password, 'must include characters from at least two groups (lower case, upper case, numbers, special characters)' unless matches >= 2
+    errors.add :password, 'must include characters from at least three groups (lower case, upper case, numbers, special characters)' unless matches >= 3
+    errors.add :password, 'must be ten characters or more in length' unless password.length >= 10
   end
 
   has_many :created_patients, class_name: 'Patient', foreign_key: 'creator_id'
 
   belongs_to :jurisdiction
 
+  # Random password for temp password changes
+  def self.rand_gen
+    SecureRandom.base58(10) + ['!', '#', '~', '='].sample + ('a'..'z').to_a.sample + ('A'..'Z').to_a.sample
```

Comment:
I think your intent is to be 'secure enough' with the call to `.base58` - but to be complete, `Array#sample` is not secure. A more secure method would be:
* `(97 + SecureRandom.random_number(26)).chr(Encoding::ASCII)` (guarantee lower)
* `(65 + SecureRandom.random_number(26)).chr(Encoding::ASCII)` (guarantee upper)
* `(33 + SecureRandom.random_number(14)).chr(Encoding::ASCII)` (guarantee symbol [`!`,`"`,`#`,`$`,`%`,`&`,`'`,`(`,`)`,`*`,`+`,`,`,`-`,`.`])

https://www.cs.cmu.edu/~pattis/15-1XX/common/handouts/ascii.html

Reply:
I like it, changed.